pkg/languagedetection/privileged_detector.go

```diff
@@ -33,17 +41,49 @@ func handleDetectorError(err error) {
 	}
 }
 
-func DetectWithPrivileges(procs []languagemodels.Process) []languagemodels.Language {
+type PrivilegedLanguageDetector struct {
+	hostProc      string
+	binaryIdCache *simplelru.LRU[binaryID, languagemodels.Language]
+	detectors     []languagemodels.Detector
+}
+
+func NewPrivilegedLanguageDetector() PrivilegedLanguageDetector {
+	lru, _ := simplelru.NewLRU[binaryID, languagemodels.Language](1000, nil) // Only errors if the size is negative, so it's safe to ignore
+
+	return PrivilegedLanguageDetector{
+		detectors:     detectorsWithPrivilege,
+		hostProc:      kernel.ProcFSRoot(),
+		binaryIdCache: lru,
+	}
+}
+
+func (l PrivilegedLanguageDetector) DetectWithPrivileges(procs []languagemodels.Process) []languagemodels.Language {
```

Comment:
I think by using value receivers, you are making copies of `PrivilegedLanguageDetector`. I think you want `func (l *PrivilegedLanguageDetector)` for most/all of your functions.

Reply:
Ah, yes you're right. Initially I believed that `DetectWithPrivileges` was small enough that it was okay to use a value receiver, but as this expands it will probably change. Updated.